File: code/R/R_Computational_Toolbox_Files/SocialNetworks_R/numPeopleEdges.R
# Callixte Nahimana
# Wofford College, Spartanburg, SC
# April, 18 2014

numPeopleEdges<-function (connPeopleMat, vertices ){
#NUMPEOPLEEDGES returns the number of edges  in a set in person-to-person graph
#   Given a person-to-person graph (connPeopleMat) and a vector of indices
#   (vertices),  the function returns the number of edges (numEdges) in the subgraph
#   with vertices.

subMat = connPeopleMat[vertices,];
numEdges = sum(sum(subMat[vertices]))/2;
return(numEdges)}

File: products/Introduction_to_Computational_Science_Modules/02_System_Dynamics/MATLAB/Files/SocialNetworksMATLAB/numPeopleEdges.m
function [ numEdges ] = numPeopleEdges( connPeopleMat, vertices )
%NUMPEOPLEEDGES returns the number of edges  in a set in person-to-person graph
%   Given a person-to-person graph (connPeopleMat) and a vector of indices
%   (vertices),  the function returns the number of edges (numEdges) in the subgraph
%   with vertices.

subMat = connPeopleMat(vertices, :);
numEdges = sum(sum(subMat(:, vertices)))/2;
end

